### ISSUE TITLE
Support "in" operator for STL containers

### ISSUE THREAD

**Issue by cmpute** [cython/cython#3652]
This is a feature request for supporting python-like "in" operator for STL containers (vector, map, etc.)

For example to test whether a set contains a value in python:
```python
a = set([1, 2])
assert 1 in a
```
But right now similar operator in Cython is not supported:
```cython
from libcpp.set cimport set
cdef set a = [1, 2]
assert 1 in a
```

This is not critically necessary but it's useful

**Comment by da-woods** [comment]
The one thing against it is that it's wrapping an operation that doesn't quite exist in the C++ STL. Obviously you can do it with something like `std::find(a.begin(), a.end(), 1) != a.end()`, but the there isn't an `std::contains` for example.

**Comment by cmpute** [comment]
Yes, but actually almost all containers support `find` function, which can be used to implement this feature which is common in python side

**Comment by scoder** [comment]
We support `for … in <c++ iterable>`, so I think supporting `in` via the same iteration mechanism is reasonable. PR welcome.

This can be implemented via a generated C++ function, one per container type. See the C++ container conversion functions in `CppConvert.pyx`, for example. This wouldn't be done through the type instance (like the container conversion), but through a function generated in `PrimaryCmpNode`, which handles the `in` and `not in` operators (and others). It can be implemented in Cython code.

**Comment by jhelgert** [comment]
Just stumbled upon this. I think it's worth mentioning that C++20 added [`.contains()`](https://en.cppreference.com/w/cpp/container/map/contains) methods for all the associated containers (map, multimap, set, multiset and the `_unordered` ones) making this less verbose.

**Comment by da-woods** [comment]
Thanks @jhelgert - I suspect practically we'd want to support C++97 still for this (it seems like kind of a basic feature to require C++20 for).

I actually think this is easier than Scoder says and it's a very small number of template functions rather than needing generated code. I'll have a look at this sometime fairly soon.
